Objects/dictobject.c

```diff
@@ -1529,39 +1529,53 @@ _PyDict_LoadGlobal(PyDictObject *globals, PyDictObject *builtins, PyObject *key)
     return value;
 }
 
-/* CAUTION: PyDict_SetItem() must guarantee that it won't resize the
- * dictionary if it's merely replacing the value for an existing key.
- * This means that it's safe to loop over a dictionary with PyDict_Next()
- * and occasionally replace a value -- but you can't insert new keys or
- * remove them.
- */
+/* Steals key and value */
 int
-PyDict_SetItem(PyObject *op, PyObject *key, PyObject *value)
+_PyDict_SetItem(PyObject *op, PyObject *key, PyObject *value)
```

Comment:
I would make the first argument `PyDictObject *mp`. It has to be a dict, so we might as well tell the C compiler that.